[warning] unused but set warning from gcc

we have a warning from gcc:
```
be/src/storage/vectorized/chunk_helper.cpp: In instantiation of ‘starrocks::vectorized::ColumnPtr starrocks::vectorized::ColumnBuilder::operator()(bool) [with starrocks::FieldType ftype = starrocks::OLAP_FIELD_TYPE_ARRAY; starrocks::vectorized::ColumnPtr = std::shared_ptr<starrocks::vectorized::Column>]’:
be/src/storage/olap_type_infra.h:120:9:   required from ‘auto starrocks::field_type_dispatch_column(starrocks::FieldType, Functor, Args ...) [with Functor = starrocks::vectorized::ColumnBuilder; Args = {bool}]’
be/src/storage/vectorized/chunk_helper.cpp:306:70:   required from here
be/src/storage/vectorized/chunk_helper.cpp:293:14: warning: variable ‘NullableIfNeed’ set but not used [-Wunused-but-set-variable]
  293 |         auto NullableIfNeed = [&](ColumnPtr col) -> ColumnPtr {
```


we need modify be/src/storage/vectorized/chunk_helper.cpp

```patch
diff --git a/be/src/storage/vectorized/chunk_helper.cpp b/be/src/storage/vectorized/chunk_helper.cpp
index 206f4b12e..1622bebf0 100644
--- a/be/src/storage/vectorized/chunk_helper.cpp
+++ b/be/src/storage/vectorized/chunk_helper.cpp
@@ -290,7 +290,7 @@ void ChunkHelper::padding_char_columns(const std::vector<size_t>& char_column_in
 struct ColumnBuilder {
     template <FieldType ftype>
     ColumnPtr operator()(bool nullable) {
-        auto NullableIfNeed = [&](ColumnPtr col) -> ColumnPtr {
+        [[maybe_unused]] auto NullableIfNeed = [&](ColumnPtr col) -> ColumnPtr {
             return nullable ? NullableColumn::create(std::move(col), NullColumn::create()) : col;
         };
```

Welcome newcomers to solve this issue

--- a/be/src/storage/vectorized/chunk_helper.cpp
+++ b/be/src/storage/vectorized/chunk_helper.cpp
@@ -290,7 +290,7 @@ void ChunkHelper::padding_char_columns(const std::vector<size_t>& char_column_in
 struct ColumnBuilder {
     template <FieldType ftype>
     ColumnPtr operator()(bool nullable) {
-        auto NullableIfNeed = [&](ColumnPtr col) -> ColumnPtr {
+        [[maybe_unused]] auto NullableIfNeed = [&](ColumnPtr col) -> ColumnPtr {
             return nullable ? NullableColumn::create(std::move(col), NullColumn::create()) : col;
         };
 

